### PR TITLE
Env: Add support for local override files.

### LINKED ISSUE #20324: @wordpress/env: Allow temporarily overriding settings in `.wp-env.json`
It is useful while working on Gutenberg to run the development environment using a different setup to the one provided by default in [Gutenberg's `~/.wp-env.json` configuration](https://github.com/WordPress/gutenberg/blob/master/.wp-env.json). 

Some examples:

- One may want to work on Core at the same time as Gutenberg by setting `"core": "~/path/to/wordpress-develop/build"`.
- One may want to test their changes against the production version of WordPress by setting `"core": null`.
- One may want to test their changes against an older version of WordPress by setting `"core": "WordPress/WordPress#5.2.0"`.
- One may want to configure their Gutenberg development environment to have some extra themes and plugins. For example, the `gutenberg-starter-theme` theme.

Here, I propose two approaches for accomplishing this. I think that we could implement one or both approaches.

## Approach 1: Command line arguments

`wp-env start` could be modified to accept command line arguments that, if specified, override the associated values that are in `.wp-env.json`.

With this approach, you could start the development server using the latest production version of Gutenberg by running:

```
wp-env start --core null
```

Starting the development server with different plugins and themes would work similarly:

```
wp-env start --plugins . elliotcondon/acf --themes WordPress/gutenberg-starter-theme
```

Considerations:

- `--core` would accept a string, and `--plugins` and `--themes` would accept an [array of strings](http://yargs.js.org/docs/#api-arraykey). 
- Specifying `--core null` will technically cause `yargs` to parse `core` as the literal string `'null'`. We'll need to modify the source parser to treat `'null'` as a special string literal which means `null`.
- These arguments should be only for _overriding_ settings. That is, it should not be possible to run `wp-env` with these command line arguments in a directory that is **not** a WordPress installation, plugin or theme; or in a directory does **not** contain a `.wp-env.json` file. 
- We will need to decide if `--plugins` and `--themes` _overrides_ the setting in `.wp-env.json` or _appends_ to the setting in `.wp-env.json`.

## Approach 2: `.wp-env.override.json`

`wp-env start` could be modified to look for a file called `.wp-env.override.json` which, if it it exists, will be merged with the `.wp-env.json` file. This is similar to [`docker-compose.override.yml`](https://docs.docker.com/compose/extends/).

Gutenberg would then add `.wp-env.override.json` to its `.gitignore` file, meaning that developers can configure their environment exactly how they like it without worrying about stashing and re-stashing changes. 

Fields in `.wp-env.override.json` would take precedence over fields in `.wp-env.json` using the same semantics as the object spread operator (`{ ...config, ...overrides }`).

That is, if one has the following `.wp-env.json`:

```json
{
	"core": "WordPress/WordPress",
	"plugins": [ "." ]
}
```

And the following `.wp-env.override.json`:

```json
{
	"plugins": [ "elliotcondon/acf" ]
}
```

The final configuration would be:

```json
{
	"core": "WordPress/WordPress",
	"plugins": [ "elliotcondon/acf" ]
}
```

---

cc. @epiqueras @noahtallen

### PR DESCRIPTION
**Pull request by epiqueras**
Closes #20324 by implementing the second approach. See https://github.com/WordPress/gutenberg/issues/20324#issuecomment-589252074.

### CODE REVIEW

File: packages/env/lib/config.js [hunk 1]
```diff
@@ -78,6 +79,18 @@ module.exports = {
...
+			overrideConfig = JSON.parse(
+				await fs.readFile(
+					configPath.replace(
+						/\.wp-env\.json$/,
+						'.wp-env.override.json'
+					),
+					'utf8'
+				)
+			);
+		} catch ( err ) {}
```
Comment (noisysocks) [inline packages/env/lib/config.js]: Let's handle the error similarly to the lines above by throwing `ValidationError` when the file can't be read or when the JSON is invalid. It's useful for troubleshooting. I for one am constantly adding trailing commas to my JSON file 🙂

**Review by noisysocks — CHANGES_REQUESTED** [review]

File: packages/env/lib/config.js (hunk 1 above)
Comment (epiqueras) [reply packages/env/lib/config.js]: Done

**Review by noisysocks — CHANGES_REQUESTED** [review]
This works great! 🌈 ✨ 

I think let's just document this new feature in `packages/env/README.md` and add an entry to `packages/env/CHANGELOG.md` and we're good to go.

**Comment by epiqueras** [comment]
Done

**Review by noisysocks — APPROVED** [review]
📈📈📈

**Pull request merged**